src/Nethermind/Nethermind.Init/Steps/EthereumStepsLoader.cs

```diff
@@ -4,71 +4,56 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
-using System.Reflection;
-using Nethermind.Api;
+using Autofac;
+using Autofac.Core;
+using Autofac.Core.Activators.Reflection;
 
 namespace Nethermind.Init.Steps
 {
     public class EthereumStepsLoader : IEthereumStepsLoader
     {
-        private readonly IEnumerable<Assembly> _stepsAssemblies;
-        private readonly Type _baseApiType = typeof(INethermindApi);
+        private readonly IEnumerable<Type> _steps;
 
-        public EthereumStepsLoader(params Assembly[] stepsAssemblies)
-            : this((IEnumerable<Assembly>)stepsAssemblies) { }
-
-        public EthereumStepsLoader(IEnumerable<Assembly> stepsAssemblies)
+        public EthereumStepsLoader(ILifetimeScope scope)
         {
-            _stepsAssemblies = stepsAssemblies;
+            _steps = scope.ComponentRegistry.RegistrationsFor(new TypedService(typeof(IStep)))
+                .Where(r => r.Activator is ReflectionActivator)
+                .Select(r => ((ReflectionActivator)r.Activator).LimitType)
```

Comment:
Can you explain? Why not other activators?

Reply:
Yea, thats a limitation here. The way the steps loader works is by checking its type. And it turns out, the easiest way to do so is by checking the activator. Unfortunately as the activator can be dynamic (see how IConfig is loaded) and of different way, this can miss some use case. Alternatively, the plugins can declare 'IStepMeta' instead of 'IStep' instead and we can use 'IStepMeta' here for a more robust solution.

Reply:
Switch to providing StepInfo instead.